packages/phpstan-rules/src/Rules/CheckClassNamespaceFollowPsr4Rule.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Symplify\PHPStanRules\Rules;
+
+use PhpParser\Node;
+use PhpParser\Node\Stmt\ClassLike;
+use PHPStan\Analyser\Scope;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+use Symplify\SmartFileSystem\SmartFileSystem;
+
+/**
+ * @see \Symplify\PHPStanRules\Tests\Rules\CheckClassNamespaceFollowPsr4Rule\CheckClassNamespaceFollowPsr4RuleTest
+ */
+final class CheckClassNamespaceFollowPsr4Rule extends AbstractSymplifyRule
+{
+    /**
+     * @var string
+     */
+    public const ERROR_MESSAGE = '%s namespace %s does not follow PSR-4 configuration in composer.json';
+
+    /**
+     * @var array<string, string>
+     */
+    private $psr4;
```

Comment:
What exactly is this property? Better name, e.g. `$autoloadPsr4Paths` would help.

```suggestion
    private $psr4 = [];
```

Reply:
updated.